In [1]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency

In [2]:
data=pd.read_csv("C:/Users/Siri/Desktop/Excelr/Assignments/Hypothesis testing/Costomer+OrderForm.csv")

In [3]:
data

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [4]:
data.describe()

Phillippines   Indonesia       Malta       India
count           300         300         300         300
unique            2           2           2           2
top      Error Free  Error Free  Error Free  Error Free
freq            271         267         269         280

In [7]:
data['Phillippines'].value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [9]:
data['Indonesia'].value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [10]:
data['Malta'].value_counts()


Error Free    269
Defective      31
Name: Malta, dtype: int64

In [11]:
data['India'].value_counts()


Error Free    280
Defective      20
Name: India, dtype: int64

In [13]:
import scipy as sp

In [14]:
chiStats = sp.stats.chi2_contingency([[271,267,269,280],[29,33,31,20]])
print('Test t=%f p-value=%f' % (chiStats[0],chiStats[1]))

Test t=3.858961 p-value=0.277102


In [16]:
# Compare p_value with α = 0.05

Inference: As (p_value = 0.2771) > (α = 0.05); Accept Null Hypthesis i.e. Independence of categorical variables Thus, customer order forms defective % does not varies by centre

Discrete data is a numerical type of data that includes whole, 
concrete numbers with specific and fixed data values determined by counting. 
Continuous data includes complex numbers and varying data values that are measured over a specific time interval

this data set is discrete data set.